ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -97,6 +97,46 @@ let networks path (list: string -> string list) =
 		|> List.map (fun (path, prefix) -> find_all_ips path prefix)
 		|> List.concat
 
+(* This function is passed the "device/vif" node and a function it can use to
+ * find the directory listing of sub-nodes. It will return a map where the
+ * keys are the xenstore paths of the VIFs' states, and the values are
+ * the corresponding state values.
+ * Example output for two vifs might be:
+ * 0/state -> 4
+ * 1/state -> 4
+ * *)
+let vif_state path (list: string -> string list) =
+	let extend base str = Printf.sprintf "%s/%s" base str in
```

Comment:
This invites a little refactoring: rather than copying this from the `extend` function above, it would be better to define it just once (just before line 58) and then call it from the three functions that use it.